packages/foam-vscode/src/dated-notes.ts

```diff
@@ -18,12 +18,15 @@ async function openDailyNoteFor(date?: Date) {
   await focusNote(dailyNotePath, isNew);
 }
 function getDailyNotePath(configuration: WorkspaceConfiguration, date: Date) {
-  const rootDirectory = workspace.workspaceFolders[0].uri.fsPath;
   const dailyNoteDirectory: string =
     configuration.get('openDailyNote.directory') ?? '.';
   const dailyNoteFilename = getDailyNoteFileName(configuration, date);
-
-  return join(rootDirectory, dailyNoteDirectory, dailyNoteFilename);
+  if (isAbsolute(dailyNoteDirectory)) {
+    return join(dailyNoteDirectory, dailyNoteFilename);
+  } else {
+    const rootDirectory = workspace.workspaceFolders[0].uri.fsPath;
```

Comment:
in connection to my comment above, we should always be _very_ wary of using the `fsPath` property anywhere outside of the `DataStore`, as it immediately makes the code platform dependent. 
`URI` is the abstraction we want to use instead in all the business logic.

Reply:
`rootDirectory` was already defined this way in the existing code. 🙈 

I will try to see if I can use `URI` for this.